Formula/payload-dumper-go.rb

```diff
@@ -0,0 +1,18 @@
+class PayloadDumperGo < Formula
+  desc "Android OTA payload dumper written in Go"
+  homepage "https://github.com/ssut/payload-dumper-go"
+  url "https://github.com/ssut/payload-dumper-go/archive/refs/tags/1.2.0.tar.gz"
+  sha256 "f0d654cba5d660c3ff8000cb510d73bc0a1eea4e725f2d79a1f16d973aae90c4"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+  depends_on "xz"
+
+  def install
+    system "go", "build", "-o", bin/"payload-dumper-go", "."
+  end
+
+  test do
+    system "#{bin}/payload-dumper-go", "-h"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Uh, that might be complicated. I'd need to either like download a full OTA from https://developers.google.com/android/ota and attempt to `payload-dumper-go -l ${file}` on it or prepare minimal payload.bin and store it somewhere.

Reply:
If downloading 821M file is ok, then I can do:
```ruby
test do
  system "curl", "-O", "https://dl.google.com/dl/android/aosp/sailfish-ota-nde63h-995d0511.zip"
  assert_match /Payload Version: 2/, shell_output("#{bin}/payload-dumper-go -l sailfish-ota-nde63h-995d0511.zip")
end
```

Reply:
Alternatively, we can go for sample_images from AOSP:
```ruby
test do
  system "curl https://android.googlesource.com/platform/system/update_engine/+/refs/tags/android-12.0.0_r16/sample_images/sample_payloads.tar.xz?format=TEXT | base64 -d | tar xJf -"
  assert_match /Payload Version: 2/, shell_output("#{bin}/payload-dumper-go -l full_payload.bin")
end
```

Reply:
full_payload.bin was small enough to b64 it in place, thus marking this is resolved.